### ISSUE TITLE
new ideas

### ISSUE THREAD

**Issue by Amin456789** [Cyberhan123/stable-diffusion-desktop#9]
Hi
i though to make a new topic to share new ideas

1.batch image save, it will be great to save all the images at once if possible
2.images have no .png or .jpg at the end of when saving them = i mean when i try to save the images i have to put .png at the end of image manually, please fix if possible
3. preset saving = it will save lots of time if we can make presets of our settings, for example when i choose taesd path and lcm sampler or etc... i can save as a preset so all i need is to load it from now on than selecting them again
4. the app sometimes is being closed by itself randomly like after saving the image

thank u so much for this lovely app

**Comment by Amin456789** [comment]
5. please add clip skip feature if possible as some models need it for better images
6. app closes when i click on generate again after i generate 1 image, please fix it if possible

can't wait for img2img also! thank u so much for everything

**Comment by JohnClaw** [comment]
> 6\. app closes when i click on generate again after i generate 1 image or aftehr batch count, please fix it if possible

I found a way how to avoid this bug. Before clicking generate button you should change RNG and Schedule in Settings.

**Comment by Cyberhan123** [comment]
This looks like a bug

**Comment by Cyberhan123** [comment]
> > 6. app closes when i click on generate again after i generate 1 image or aftehr batch count, please fix it if possible
> 
> I found a way how to avoid this bug. Before clicking generate button you should change RNG and Schedule in Settings.

Ok，I need test full.

**Comment by Cyberhan123** [comment]
@Amin456789  The crash of the program is related to this problem. https://github.com/leejet/stable-diffusion.cpp/issues/146. I have created a related issue and currently disabled this parameter in the app.

**Comment by Amin456789** [comment]
thank u for this

**Comment by Cyberhan123** [comment]
I'm sending an emergency fix now, and will improve the picture and settings issues later, they are very simple

**Comment by Amin456789** [comment]
hmm it seems it doesnt use full cpu anymore and takes double to make images, so we have to sd cpp fix this first,

**Comment by Amin456789** [comment]
i think i know where is the problem = everytime we make changes to settings it will apply and reload the model and that is how it works again, i think u should revert back to that parameter thing cyberhan and make ur gui to offload the model when generation is done and reload it again whenever we are about to generate again, this is how sd cpp does it too, model must be reloaded
a simple gpu enabled and disabled does the trick and reloads the model

**Comment by Amin456789** [comment]
new update = just disabling free params immideatly is enough, johnclaw just disable that in prevois version and it wont hang again,. so the problem is that

**Comment by Amin456789** [comment]
7. it will be great if prompt section can go over 75 token too, 150 is good i think
8. my laptop screen is 1366x768, i can barely see batch count, could u please somehow make the ui more comfortable with laptop screens like mine, thnx

**Comment by Cyberhan123** [comment]
> 7. it will be great if prompt section can go over 75 token too, 150 is good i think
> 8. my laptop screen is 1366x768, i can barely see batch count, could u please somehow make the ui more comfortable with laptop screens like mine, thnx

This is not my limit, you can see it is the clip `modle` limit of sd itself

**Comment by Cyberhan123** [comment]
In other words, if you exceed it, you will be ignored.

**Comment by Amin456789** [comment]
oh that's why in the past my images mostly were bad as it was ignoring my words when i was over 75, thank u for the answer

**Comment by Amin456789** [comment]
hey cyberhan, i see u merged the new update, could u please release the exe in releases section too as i cant compile it myself.
it seems u fixed everything in the new update so im gonna close this if its ok with u,

btw, iv read ur oigin.md, thank u so much for making this app, this is my favroite gui right now and i deleted every sd webui and only using this, thank u for mentioning me and please keep up the good work, this app will rock in the future, leejet should add this to the read me in his page, please ask him

**Comment by Cyberhan123** [comment]
There is a frustrating problem. Although I found the reason, I am weighing how to deal with it. VaeDecodeOnly can be turned on and off in txt2img, but in img2img mode, it must remain turned off, otherwise the program will crash. The problem lies in table-diffusion There is currently no API for setting options, otherwise we need to reload the model and re-initialize it again, so I plan to submit a PR to modify the current logic of stable-diffusion.cpp

**Comment by Cyberhan123** [comment]
By the way, I recommend you keep other webui that you don't need to use just this one, it stresses me out a lot and I'm doing this in my spare time.

**Comment by Amin456789** [comment]
thank u for ur hard work, please dont stress out, do it any time u can, even if it takes months to fix something its all great, i though u fixd and merged it that is why i asked for it, otherwise pelase dont hurry,

**Comment by Amin456789** [comment]
tnx for the new update but i think img2img is broken, it doesnt matter what setting i choose, it always gives me the same image i gave it to it excacly, and sometimes it even jump out of the program

**Comment by Amin456789** [comment]
another bug is when i choose fp16 in always makes the fp16 model fp32, and i have load it once then go to the app and rechoose so it can load it in fp16 again, it will be great it automaticly always choose the weight's type like in sd cpp

**Comment by Amin456789** [comment]
what about u johnclaw, please report all bugs

**Comment by Cyberhan123** [comment]
> tnx for the new update but i think img2img is broken, it doesnt matter what setting i choose, it always gives me the same image i gave it to it excacly, and sometimes it even jump out of the program like using batch count, i am using lcm dreamshaper 7

@Amin456789 
Can you give me a specific example? What model did you use to set what parameters? Automatically recommend `wtype`. I found this also problem, but I need to wait for the ci upgrade of sd.cpp to solve it. Because the type enumeration has been increased.

**Comment by Cyberhan123** [comment]
Another thing to note is that the parameters are very important, because the `simple methods` of some models can only output correctly under the correct parameters .

**Comment by Amin456789** [comment]
i tested with lcm dreamshaper 7 and euler a and lcm sampler and euler also tested i think, and i put on 4 to 8 steps if i remember correctly and cfg 1 and 8 also i tested in many ways and it didnt work, i hope it get fixed with ur ci update for sd cpp

**Comment by Amin456789** [comment]
btw cyberhan, tnx for all ur efforts, together with u and ur fans we will make this the best sd cpp out there, this is the future of sd as one single small app is the best and easier to use. long journey ahead with lots of new featuers such as svd and inainting outpinting, btw u should post ur app in stable diffusion section on reddit to people find it

**Comment by Cyberhan123** [comment]
I want to solve all the problems first and make this software really usable，Yeah I guess I overlooked this setting in my rush to publish, thanks.

**Comment by Cyberhan123** [comment]
I think I need to add some interactive experiences:
1. The same model directory settings, unified model path
2. Can be loaded and released later.
3. Can set the theme according to your own needs
4. Model download page
These are already planned and will be coming soon I think

**Comment by Amin456789** [comment]
great, thank u

**Comment by Amin456789** [comment]
1. whenever we switch to text to img and img2img the model reloads again = please make it to not reload as sometimes it takes time specially with sdxl turbo as in the future when u fix vaedecode only the we can wait i guess
2. i tested taesd and taesdxl and both works great, please add them to the feature as u forgot to remove taesd from todo list

**Comment by Cyberhan123** [comment]
Yes, this is related to the API design of sd.cpp. I have also related related enhancement issues.

**Comment by Amin456789** [comment]
great update, img2img works now thank u
1. the settings change are like choosing fp type needs a restart of the app or we need to load the model and make the changes and load the model again to take the new fp effect [sdxl turbo needs to be 5.1 for me to work on my laptop]. not a deal breaker but it will be great if the app take the setting changes the first time we make changes
2. it will be great if u adjust the app size for laptops with 1366x768, i need to use an app named altdrag to drag the app higher to see the bottom of the app as i can't see it, or maybe u can just make the app that when we maximize it it shows all the options and whole app if possible

tnx again, great update,  i like the auto save settings too,

**Comment by Cyberhan123** [comment]
The current problem is that the option cannot be displayed when using 768px height. This is a UI design issue. Do you have any ideas?

**Comment by Amin456789** [comment]
hmm no idea but there must be a way, as there are some apps when u maximisize the window the app displays every items even the screen is smaller than the app, or u can overhaul the app a bit more to fit with that size, but it is ok cyberhan don't mind it please, work on more important stuff and in the time something will spark and you can fix it then

**Comment by Amin456789** [comment]
hey guys im back with new ideas=

1. cyberhan, svd as u know its being slowly implented i thogh it will be great that u start to add the parametres and the section of svd in ur app slowly when svd is ready
2. autosave of images and image location folder = it will be great if there is a seciton we choose for the image place in settings that the app auto save all images in there after images are generated as saving lots of images manually is lots of time consuming, also it will be great if the saved images get the prompt as the image name and put a number in front of the name if there is a batch image
3.  a button to reset all saved settings as there is no way i think to remove selected taesd if we dont wanna use it
4. instead of choosing models we can select a folder for the models and the app can detect the models and list them so all we have to do is to select them via app

**Comment by Cyberhan123** [comment]
I also discovered it, but since it’s Chinese New Year, I don’t have much time to do it.

**Comment by Amin456789** [comment]
happy new year, hope u have a great year ahead, its all ok, make the new changes when the time is right

**Comment by Cyberhan123** [comment]
I fixed the inability to clear VAE, and other settings that need to be cleared. Other functions cannot be implemented now because our company has laid off employees. Unfortunately, I am one of them. I am looking for a new full-time job until everything is stable. When the time comes, I will devote my free time to this project. Sorry to everyone who was looking forward to this project.
